src/storage/v2/id_types.hpp

```diff
@@ -50,10 +50,10 @@ namespace memgraph::storage {
   inline bool operator<=(const name &first, const name &second) { return first.AsUint() <= second.AsUint(); } \
   inline bool operator>=(const name &first, const name &second) { return first.AsUint() >= second.AsUint(); }
 
-STORAGE_DEFINE_ID_TYPE(Gid);
-STORAGE_DEFINE_ID_TYPE(LabelId);
-STORAGE_DEFINE_ID_TYPE(PropertyId);
-STORAGE_DEFINE_ID_TYPE(EdgeTypeId);
+STORAGE_DEFINE_ID_TYPE(Gid, uint64_t, int64_t, utils::ParseStringToUint64);
```

Comment:
can you check maybe name_id_mapper functionality ? do we have any risk of running that because every time someone makes a typo the name id mapper is still going to autoincrement the value?
also, do we want separate counters for every of them (separate for label, property and edge type) ?

Reply:
Ok sure `NameToId` is not const, so even in non-write based queries we can store and bump the id. 
That should be changed, a const version is needed, which would return std::nullopt.

We should keep separate counters.

Reply:
I would fix the NameToId issue in another PR, if and when it become a priority/likely issue.

It is possibly to saturate the mappings with typos, but as it stands right now maybe not a priority.